### PR TITLE
Basic CRUD operations for deck and flashcard

### PR DESCRIPTION
**Pull request by huiningd**

### CODE REVIEW

File: src/main/kotlin/com/example/flashcardbackend/deck/DeckRepository.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        val sql = """
+            SELECT deck.*, flashcard.id AS flashcard_id, flashcard.deck_id 
+            FROM deck, flashcard 
+            WHERE deck.id = flashcard.deck_id AND flashcard.id = ?
+        """.trimIndent()
+        return jdbcTemplate.query(sql, DeckResultSetExtractor(), id)
+    }
+}
+
+class DeckResultSetExtractor : ResultSetExtractor<Deck?> {
```
Comment (petrikainulainen-nitor) [inline src/main/kotlin/com/example/flashcardbackend/deck/DeckRepository.kt:64]: I just remembered that here is a library called SimpleFlatMapper (https://simpleflatmapper.org/) that should help you get rid of this result set extractor. It's essentially a mapper library that helps you implement read-only one-to-many relationships. I have used it in the past, so, if you run into any problems, let me know. Also, you can fix this in a separate PR.

File: src/main/kotlin/com/example/flashcardbackend/deckgroup/DeckGroupRepository.kt [hunk 2]
```diff
@@ -1,55 +1,83 @@
...
-        return null
+        val sql = """
+            SELECT deck_group.*, deck.* 
+            FROM deck_group, deck 
+            WHERE deck_group.id = deck.group_id AND deck_group.id = ?
+        """.trimIndent()
+        return jdbcTemplate.query(sql, DeckGroupResultSetExtractor(), id)
+    }
+}
+
```
Comment (petrikainulainen-nitor) [inline src/main/kotlin/com/example/flashcardbackend/deckgroup/DeckGroupRepository.kt:60]: Like I mentioned earlier, the SimpleFlatMapper should help you get rid off this resultset extractor.

File: src/main/kotlin/com/example/flashcardbackend/deck/DeckRepository.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.dao.DataAccessException
+import org.springframework.jdbc.core.JdbcTemplate
+import org.springframework.jdbc.core.ResultSetExtractor
+import org.springframework.stereotype.Repository
+import org.springframework.transaction.annotation.Transactional
+import java.sql.ResultSet
+import java.sql.SQLException
+
+@Transactional
```
Comment (petrikainulainen-nitor) [inline src/main/kotlin/com/example/flashcardbackend/deck/DeckRepository.kt]: This is fine, but my OCD tells me that these annotations must be in this order: @Repository @Transactional 🙈

**Review by petrikainulainen-nitor — APPROVED** [review]
I approve this PR like we agreed on the meeting. You can fix the things we discussed in new PRs. Also, take a look at the SimpleFlatMapper library. It's quite useful when you want to implement read-only one-to-many mappings in SQL queries without using JPA.

File: src/main/kotlin/com/example/flashcardbackend/deck/DeckRepository.kt (hunk 1 above)
Comment (huiningd) [reply src/main/kotlin/com/example/flashcardbackend/deck/DeckRepository.kt:64]: Thank you! I will look into SimpleFlatMapper!

File: src/main/kotlin/com/example/flashcardbackend/deck/DeckRepository.kt (hunk 3 above)
Comment (huiningd) [reply src/main/kotlin/com/example/flashcardbackend/deck/DeckRepository.kt]: I should add ktlint to the project

File: src/main/kotlin/com/example/flashcardbackend/deck/DeckRepository.kt (hunk 1 above)
Comment (huiningd) [reply src/main/kotlin/com/example/flashcardbackend/deck/DeckRepository.kt:64]: Yeah, I think I will fix it in a separate PR

**Pull request merged**